src/model/RefrigerationSystem.cpp

```diff
@@ -189,26 +190,37 @@ namespace detail {
       result.insert(result.end(), removedTransferLoads.begin(), removedTransferLoads.end());
     }
 
+    // We're clearing the compressor/HighStage compressor list objects first. otherwise a compressor on the high stage list will call
+    // RefrgerationCompressor_Impl::system which will try to locate the compressorList which was deleted first
+    for (auto& c: compressors()) {
+      std::vector<IdfObject> removedCompressor = c.remove();
+      result.insert(result.end(), removedCompressor.begin(), removedCompressor.end());
+    }
+    for (auto& c: highStageCompressors()) {
+      std::vector<IdfObject> removedCompressor = c.remove();
+      result.insert(result.end(), removedCompressor.begin(), removedCompressor.end());
+    }
     if (boost::optional<ModelObjectList> compressorList = this->compressorList()) {
       std::vector<IdfObject> removedCompressors = compressorList->remove();
       result.insert(result.end(), removedCompressors.begin(), removedCompressors.end());
     }
+    if (boost::optional<ModelObjectList> highStageCompressorList = this->highStageCompressorList()) {
+      std::vector<IdfObject> removedHighStageCompressors = highStageCompressorList->remove();
+      result.insert(result.end(), removedHighStageCompressors.begin(), removedHighStageCompressors.end());
+    }
 
+    // TODO: technically this object can be shared accross multiple systems..
     if ( boost::optional<RefrigerationSubcoolerMechanical> mechSubcooler = this->mechanicalSubcooler() ) {
```

Comment:
Should be disallow sharing `RefrigerationSubcoolerMechanical` accross several RefrigerationSystem? Or not delete it unless it becomes orphaned? @kbenne ?

Reply:
im generally not a fan of sharing in cases like this, but if it is shared it seems like it would be a ResourceObject. Recall as an example Curve which recently became shared and a ResourceObject. In this case if it is not currently shared in the implementation I would be ok with merging as is, unless someone else makes an argument to share these objects.

Reply:
It is shared currently, see the code example above.

If I add the same Subcooler to two RefrigerationSystem, and delete one RefrigerationSystem, then the other suddenly doesn't have any Subcooler anymore.

Options :
1. In RefrigerationSystem::remove: 
   1.1. check if subcooler ends up orphaned, then delete if that's the case
   1.2 do not remove anyways
2. In RefrigerationSystem::addSubcooler: enforce unicity.

I would go for 2. myself, for consistency with the rest of the object as well. I would accept falling back to 1.1. and I dislike 1.2

Reply:
2 is good with me too. 1 (1.2 especially) sounds like ResourceObject and if you go that direction then I think the base class should be officially changed.